global-script.lua

```diff
@@ -2757,12 +2799,30 @@ function upCastPosSizRot(oPos,size,rot,dist,multi,tags)
 end
 ---- Block Square Section
 function setupPlayerArea(params)
-    --Sets position/color for the button, spawns it
-    params.obj.createButton({
+    -- Figure out what color we're supposed to be, or if playerswapping is even allowed.
+    local obj = params.obj
+    obj.createButton({
         label="Energy Cost: 0", click_function="nullFunc",
         position={0,2.24,-11.2}, rotation={0,180,0}, height=0, width=0,
         font_color={1,1,1}, font_size=500
     })
+
+    local color
+    for k, v in pairs(playerBlocks) do
+        if v.guid == obj.guid then
+            color = k
+            break
+        end
+    end
+    obj.setVar("playerColor", color)  -- May be nil
+    if color then
```

Comment:
you could add a check to make sure that `gameStarted` is false, which should alleviate the need to store playerBlocks in the save data

Reply:
`playerBlocks[color]` is also nil when a spirit is chosen by that color, so it is possible that some -- but not all slots are cleared out.  (This is used to signal onload which slots get a 'Move to here' button).

It is true this can be optimized out if gameStarted, but for code that executes infrequently (6x on load/undo/etc.) and has a very small amount of save data in a live game (a single empty table), I felt the reduced code complexity was more valuable here.